src/Sentry.AspNetCore/SentryTracingMiddleware.cs

```diff
@@ -161,10 +160,26 @@ public async Task InvokeAsync(HttpContext context)
 
                     var status = SpanStatusConverter.FromHttpStatusCode(context.Response.StatusCode);
 
-                    // If no Name was found for Transaction, fallback to UnknownRoute name.
+                    // If no Name was found for Transaction, then we don't have the route.
                     if (transaction.Name == string.Empty)
                     {
-                        transaction.Name = UnknownRouteTransactionName;
+                        var method = context.Request.Method.ToUpperInvariant();
+
+                        // If we've set a TransactionNameProvider, use that here
+                        var customTransactionName = context.TryGetCustomTransactionName();
+                        if (!string.IsNullOrEmpty(customTransactionName))
```

Comment:
can we invert this to avoid the negation

Reply:
I'd prefer not to on this one, as it keeps it in the sequence that they would apply.